src/pyhf/pdf.py

```diff
@@ -205,24 +318,13 @@ def __init__(self, spec, batch_size=None, **config_kwargs):
             self.config, batch_size=self.batch_size
         )
 
-        self._factor_mods = [
-            modtype
-            for modtype, mod in modifiers.uncombined.items()
-            if mod.op_code == 'multiplication'
-        ]
-        self._delta_mods = [
-            modtype
-            for modtype, mod in modifiers.uncombined.items()
-            if mod.op_code == 'addition'
-        ]
-
         self.viewer_aux = ParamViewer(
             (self.batch_size or 1, len(self.config.suggested_init())),
             self.config.par_map,
             self.config.auxdata_order,
         )
 
-    def _create_nominal_and_modifiers(self):
+    def _create_nominal_and_modifiers(self, config, spec):
```

Comment:
why did this get refactored to pass in config/spec?

Reply:
I would like this to be at some point a "factory method" either as a `classmethod` or `staticmethod`..together with`_ModelConfig()`


if you look at the init we're almost at the point where

```
        self.config = _ModelConfig(self.spec, **config_kwargs)
        mega_mods, _nominal_rates = self._create_nominal_and_modifiers(
            self.config, self.spec
        )
        self.main_model = _MainModel(
            self.config,
            mega_mods=mega_mods,
            nominal_rates=_nominal_rates,
            batch_size=self.batch_size,
        )

       ... snip ...

        self.constraint_model = _ConstraintModel(
            config=self.config, batch_size=self.batch_size
        )
```

I'd like to be able to have multiple ctors .. one being from a JSON spec

Reply:
note that `self` is not used at all in the body

Reply:
<img width="515" alt="screenshot" src="https://user-images.githubusercontent.com/2318083/64441741-b92bb300-d0ce-11e9-9d84-32cab21b9c46.png">

i.e. not how in`Model` all but `_create_nominal_and_modifiers` are either temporary forwarding functions to the  constituent models or functions that we consider to be part of the emerging statisfactory model  api